probatus/metric_volatility/metric.py

```diff
@@ -25,34 +25,53 @@
-        scorers: (list of Scorers) list of Scorer objects used to score the trained model
-        train_sampling_type: (Optional str) string indicating what type of sampling should be applied on train set:
-                - None indicates that no additional sampling is done after splitting data
-                - 'bootstrap' indicates that sampling with replacement will be performed on train data
-                - 'subsample': indicates that sampling without repetition will be performed  on train data
-        test_sampling_type: (Optional str) string indicating what type of sampling should be applied on test set:
-                - None indicates that no additional sampling is done after splitting data
-                - 'bootstrap' indicates that sampling with replacement will be performed on test data
-                - 'subsample': indicates that sampling without repetition will be performed  on test data
-        train_sampling_fraction: (Optional float): fraction of train data sampled, if sample_train_type is not None.
-                Default value is 1
-        test_sampling_fraction: (Optional float): fraction of test data sampled, if sample_test_type is not None.
-                Default value is 1
+        X (np.array or pd.DataFrame):
+            Dataset with features.
 
-    Returns: 
-        (pd.Dataframe) Dataframe with results for a given model trained. Rows indicate the metric measured and columns
-         ther esults
+        y (np.array or pd.Series):
+            Target of the prediction.
+
+        clf (model object):
+            Binary classification model or pipeline.
+
+        test_size (float):
+            Fraction of data used for testing the model.
+
+        split_seed (int):
+            Randomized seed used for splitting data.
+
+        scorers (list of Scorers):
+            List of Scorer objects used to score the trained model.
+
+        train_sampling_type (str, optional):
+            String indicating what type of sampling should be applied on train set:
 
+                - `None`: indicates that no additional sampling is done after splitting data,
+                - `'bootstrap'`: indicates that sampling with replacement will be performed on train data,
+                - `'subsample'`: indicates that sampling without repetition will be performed  on train data.
+
+        test_sampling_type (str, optional):
+            string indicating what type of sampling should be applied on test set:
+
+                - `None`: indicates that no additional sampling is done after splitting data
+                - `'bootstrap'`: indicates that sampling with replacement will be performed on test data
+                - `'subsample'`: indicates that sampling without repetition will be performed  on test data
+
+        train_sampling_fraction (float, optional):
+            Fraction of train data sampled, if sample_train_type is not None. Default value is 1.
```

Comment:
Bootstrapping can also be done with sample size smaller or bigger than 1. Which means that you might want to oversample or undersample the data.

I would keep this one for now. I agree this would be slightly easier to use with the different default, but the problem is if you have a default of subsample, then the subsample_fraction cannot be 1, because it will throw an error. So for many users the easiest way to start with the class is like this.